File: src/main/java/com/example/probation/validatiion/annotation/Password.java
package com.example.probation.validatiion.annotation;

import com.example.probation.validatiion.validator.PasswordValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Target({ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Constraint(validatedBy = PasswordValidator.class)

public @interface Password {
    String message() default "{password.invalid}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/com/example/probation/validation/annotation/Password.kt
package com.example.probation.validation.annotation

import com.example.probation.validation.validator.PasswordValidator
import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
@Constraint(validatedBy = [PasswordValidator::class])
annotation class Password(
    val message: String = "{password.invalid}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
